server/service/integration_mdm_test.go

```diff
@@ -402,30 +418,25 @@ func (s *integrationMDMTestSuite) TestProfileManagement() {
 	}
 
 	// trigger a profile sync
-	_, err = s.profileSchedule.Trigger()
-	require.NoError(t, err)
-	wg.Wait()
+	triggerSchedule()
 
 	installs, removes := checkNextPayloads()
-	// verify that we received both profiles
-	require.ElementsMatch(t, installs, globalProfiles)
+	// verify that we received all profiles
+	require.ElementsMatch(t, installs, append(globalProfiles, fleetdProfile.Bytes()))
 	require.Empty(t, removes)
 
 	// add the host to a team
 	err = s.ds.AddHostsToTeam(ctx, &tm.ID, []uint{host.ID})
 	require.NoError(t, err)
 
 	// trigger a profile sync
-	wg.Add(3)
-	_, err = s.profileSchedule.Trigger()
-	require.NoError(t, err)
-	wg.Wait()
+	triggerSchedule()
 
 	installs, removes = checkNextPayloads()
 	// verify that we should install the team profile
 	require.ElementsMatch(t, installs, teamProfiles)
 	// verify that we should delete both profiles
-	require.ElementsMatch(t, removes, []string{"I1", "I2"})
+	require.ElementsMatch(t, removes, []string{"I1", "I2", mobileconfig.FleetdConfigPayloadIdentifier})
```

Comment:
one thing this test made me realize: if the host changes teams, we're going to remove the profile. A fix for this could be to add a conditional to the SQL statement in `fleet.Datastore.ListMDMAppleProfilesToRemove`, or an `if/else` in the cron to skip this profile.

Do you folks have any thoughts?

Reply:
There's also the case when the status is set to NULL to indicate it is pending, so it's more than just the cron. THinking out loud, I wonder if we should introduce an `immutable` column (or something like that) in the host profiles table and mobile config profiles table (so that we know to set the immutable to true in the host profiles table once installed), and just ignore from the diffs any profiles that have this column to true?

But then again, it's trickier than that, as we want to report it as missing if it's in the team/no team list but not on the host, except if the host already has that profile type from another team/no team, right?

Reply:
> There's also the case when the status is set to NULL to indicate it is pending, so it's more than just the cron. 

you mean that we probably don't want to show this profile as pending?

> But then again, it's trickier than that, as we want to report it as missing if it's in the team/no team list but not on the host, except if the host already has that profile type from another team/no team, right?

hmm thinking though this, if the host switches teams there are two possibilities:

1. The team also has a `com.fleetdm.fleetd.config` profile, in which case ideally we want to remove the profile from the old team + install the profile from the new team
2. The team doesn't have a `com.fleetdm.fleetd.config` profile, in which case we keep the profile

as you mention... trickier

Another bit to keep in mind is that, I _think_ there's a real chance in the medium-term future that we'll want to target profiles to "groups of hosts" using something else than a team, for example: "install this profile in all M1 macs" (probably using an osquery query)

---

I wonder: another thing I have considered is to just send a one-off `InstallProfile` command when the host enrolls instead of leveraging the logic of the cron. This has some advantages:

- The profile is only installed on hosts that really need it
- We don't have to deal with teams

The drawback is that we kind of lose track of that profile, but maybe it's something we can tackle as part of #9780 ?

Reply:
> The team doesn't have a com.fleetdm.fleetd.config profile, in which case we keep the profile

Are we sure this is what the expected UX for this scenario rather than switching to a "no team" profile?

Reply:
we decided to add this profile to all teams, this way we don't have to give it special treatment.